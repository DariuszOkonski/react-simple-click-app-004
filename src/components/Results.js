import React from 'react';

const Results = (props) => {
  return (
    <React.Fragment>
      <h2 id='first'>Number of clicks: {props.state.clicks}</h2>
      <h2 id='second'>{props.state.clicks > 10 ? <span>CPU overheating</span> : '----------------------'}</h2>
      <h2 id='third'>Result: {props.state.result}</h2>
    </React.Fragment>
  );
}

export default Results;