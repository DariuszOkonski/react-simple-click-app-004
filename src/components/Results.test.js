import React from 'react';
import ReactDOM from 'react-dom';
import Results from './Results';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Results tests', () => {

  it('Results renders without a problem', () => {
    const state = {
      clicks: 0,
      result: 0,
    }
    const div = document.createElement('div');
    ReactDOM.render(<Results state={state} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot matches', () => {
    const state = {
      clicks: 0,
      result: 0,
    }
    const wrapper = shallow(<Results state={state} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('h2#first renders', () => {
    const state = {
      clicks: 1,
      result: 0,
    }
    const wrapper = shallow(<Results state={state} />);

    expect(wrapper.find('h2#first').text()).toBe('Number of clicks: 1');
  });

  it('h2#second renders, clicks less than 10', () => {
    const state = {
      clicks: 1,
      result: 0,
    }
    const wrapper = shallow(<Results state={state} />);

    expect(wrapper.find('h2#second').text()).toBe('----------------------');
  });

  it('h2#second renders, clicks more than 10', () => {
    const state = {
      clicks: 11,
      result: 0,
    }
    const wrapper = shallow(<Results state={state} />);

    expect(wrapper.find('h2#second').text()).toBe('CPU overheating');
  });

  it('h2#third renders', () => {
    const state = {
      clicks: 11,
      result: 2,
    }

    const wrapper = shallow(<Results state={state} />);

    expect(wrapper.find('h2#third').text()).toBe('Result: 2');
  });

});