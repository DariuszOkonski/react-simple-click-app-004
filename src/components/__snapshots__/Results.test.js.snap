// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results tests Snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Results
    state={
      Object {
        "clicks": 0,
        "result": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h2
          id="first"
        >
          Number of clicks: 
          0
        </h2>,
        <h2
          id="second"
        >
          ----------------------
        </h2>,
        <h2
          id="third"
        >
          Result: 
          0
        </h2>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Number of clicks: ",
            0,
          ],
          "id": "first",
        },
        "ref": null,
        "rendered": Array [
          "Number of clicks: ",
          0,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "----------------------",
          "id": "second",
        },
        "ref": null,
        "rendered": "----------------------",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Result: ",
            0,
          ],
          "id": "third",
        },
        "ref": null,
        "rendered": Array [
          "Result: ",
          0,
        ],
        "type": "h2",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h2
            id="first"
          >
            Number of clicks: 
            0
          </h2>,
          <h2
            id="second"
          >
            ----------------------
          </h2>,
          <h2
            id="third"
          >
            Result: 
            0
          </h2>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Number of clicks: ",
              0,
            ],
            "id": "first",
          },
          "ref": null,
          "rendered": Array [
            "Number of clicks: ",
            0,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "----------------------",
            "id": "second",
          },
          "ref": null,
          "rendered": "----------------------",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Result: ",
              0,
            ],
            "id": "third",
          },
          "ref": null,
          "rendered": Array [
            "Result: ",
            0,
          ],
          "type": "h2",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
